tools/autograd/gen_variable_type.py

```diff
@@ -769,47 +771,26 @@ def wrap_output(call):
             # See NOTE [ Autograd View Variables ] in variable.h for details.
             differentiable_output_vars = {r['name'] for r in differentiable_outputs}
             tensor_output_vars = {r['name'] for r in returns if 'Tensor' in r['type']}
-            if not isinstance(view_info, dict):
-                if len(differentiable_output_vars) == len(tensor_output_vars):
-                    # all outputs are differentiable
-                    return 'as_view({}, {}, true)'.format(view_info, call), []
-                elif len(differentiable_output_vars) == 0:
-                    # no output is differentiable
-                    return 'as_view({}, {}, false)'.format(view_info, call), []
-                else:
-                    # some of the outputs are differentiable
-                    # need to expand to dict mode, i.e., one entry per output
-                    base_name = view_info
-                    view_info_dict = {}
-                    for i, return_info in enumerate(returns):
-                        if 'Tensor' in return_info['type']:
-                            view_info_dict[i] = base_name
+
+            if not isinstance(view_info, str):
+                raise TypeError("The view info should be a string for {}".format(base_name))
+
+            if len(differentiable_output_vars) == 0:
+                # no output is differentiable
+                return 'as_view({}, {}, false)'.format(view_info, call), []
+            elif len(differentiable_output_vars) == 1:
+                # Single differentiable output
+                extra = []
+                is_differentiable_view = 'true'
+                wrapped_call = 'as_view({}, {}, {})'.format(view_info, call, is_differentiable_view)
+                return_info = returns[0]
+                if return_info['type'] != 'Tensor':
```

Comment:
nit: A more defensive way to code this test is to explicitly test for `Tensor` and `TensorList`, and raise an assert failure if the return info matches neither of these. If someone adds support for a new kind of differentiable output (or introduces a bug where a non-Tensor type is marked differentiable), they will then get an error at codegen time, as opposed to a missing overload error at compile time.